src/Provider.tsx

```diff
@@ -1,17 +1,21 @@
-/* eslint-disable react/prop-types */
-import React from "react"
+import * as React from "react"
 import { shallowEqual } from "./utils/utils"
+import { IValueMap } from "./inject"
 
-export const MobXProviderContext = React.createContext({})
+export type TProviderShape = Record<string, any>
+export const MobXProviderContext: React.Context<TProviderShape> = React.createContext({})
```

Comment:
Let's drop `TProviderShape` and utilize `IValueMap` everywhere.
```suggestion
export const MobXProviderContext: React.Context<IValueMap> = React.createContext({})
```